caikit/interfaces/nlp/data_model/text.py

```diff
@@ -59,3 +59,16 @@ class TokenizationStreamResult(TokenizationResults):
         int, FieldNumber(2)
     ]  # Result index up to which text is processed
     start_index: Annotated[int, FieldNumber(3)]  # Result start index for processed text
+
+
+@dataobject(package=NLP_PACKAGE)
+class ChunkerTokenizationStreamResult(TokenizationStreamResult):
+    """
+    Streaming tokenization result that provides pointer to the input chunk processed
+    """
+
+    # Below 2 represent pointer from the input stream.
+    # These are different from start and processed_index index returned from
+    # TokenizationStreamResult, which refers to the char span
+    input_start_index: Annotated[int, FieldNumber(5)]
```

Comment:
Small thought: It might be good to use a fully different set of field numbers for derived messages like this so that if additional fields are added to the parent, they don't need to be aware of all children. I'd suggest going with something like `20` and `21` or something like that.

Reply:
good point.. otherwise, there may be cases where parent adds new field and now things are overlapping